### PR TITLE
Merge rrweb, rrweb-snapshot, rrweb-player into a mono repository

### PR DESCRIPTION
**Pull request by Mark-Fenng**
1. In this PR, I use [yarn workspaces](https://classic.yarnpkg.com/en/docs/workspaces/) to link multiple packages.
2. Update Travis-CI config.

Related issue #582

### CODE REVIEW

**Comment by Yuyz0112** [comment]
Short questions before looking into the PR:

1. Is this only works when developers using yarn instead of npm?
2. Is it possible to keep the git history of snapshot and player repo after moving these files.

**Comment by Mark-Fenng** [comment]
1. I think if developers want to link three repos, yarn or npm(version>=7) is required.
2. I did a little research about merge of git history and it seems like sub-tree merge is good approach.

**Comment by Yuyz0112** [comment]
Not quite sure how does lerna does the link things.

**Comment by Mark-Fenng** [comment]
In the doc of yarn workspaces, it says yarn workspaces provide the basic support for linking packages and lerna has much more abilities. 
According to the discussion in the issue, I suppose workspaces can meet our requirement for now.

**Comment by Juice10** [comment]
You might also want to consider [pnpm](https://pnpm.io) instead of Lerna, I don't have a lot of experience with it but it seems like it would simplify things: https://www.takeshape.io/articles/why-we-switched-from-yarn-to-pnpm/

**Comment by Mark-Fenng** [comment]
@Yuyz0112 How do you like the choice of package manager? Workspaces, lerna or pnpm

**Comment by Mark-Fenng** [comment]
This PR is almost done except for a build warning.
Tips:
1. Because we want to keep the commit history of the other two projects, the number of commits in this PR may be a little too much. And because of the same reason, when merging this PR, we should __avoid the squash operation__.
2. I think we can keep using yarn workspace as our package manager for now and it's not hard to replace it with another manager if yarn can't meet our requirement later.
3. There's still a build warning as follows in the rrweb-player package.
```
(!) Unresolved dependencies
[https://rollupjs.org/guide/en/#warning-treating-module-as-external-dependency](https://rollupjs.org/guide/en/#warning-treating-module-as-external-dependency)
rrweb (imported by src/Player.svelte, src/Controller.svelte)
(!) Missing global variable name
Use output.globals to specify browser global variable names corresponding to external modules
rrweb (guessing 'rrweb')
created public/bundle.js in 1.8s
```
It seems like its rollup config doesn't fit for monorepo environment and any help would be great.

File: packages/rrweb-snapshot/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# rrweb-snapshot
+
+[![Build Status](https://travis-ci.org/rrweb-io/rrweb-snapshot.svg?branch=master)](https://travis-ci.org/rrweb-io/rrweb-snapshot) [![Join the chat at https://gitter.im/rrweb-io/rrweb-snapshot](https://badges.gitter.im/rrweb-io/rrweb-snapshot.svg)](https://gitter.im/rrweb-io/rrweb-snapshot?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
```
Comment (Juice10) [inline packages/rrweb-snapshot/README.md:3]: Build status icon should be changed to this repo
```suggestion
[![Build Status](https://travis-ci.org/rrweb-io/rrweb.svg?branch=master)](https://travis-ci.org/rrweb-io/rrweb) [![Join the chat at https://gitter.im/rrweb-io/rrweb-snapshot](https://badges.gitter.im/rrweb-io/rrweb-snapshot.svg)](https://gitter.im/rrweb-io/rrweb-snapshot?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
```

File: packages/rrweb-snapshot/package.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  "files": [
+    "dist",
+    "lib",
+    "es",
+    "typings"
+  ],
+  "author": "yanzhen@smartx.com",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/rrweb-io/rrweb-snapshot/issues"
```
Comment (Juice10) [inline packages/rrweb-snapshot/package.json:34]:
```suggestion
    "url": "https://github.com/rrweb-io/rrweb/issues"
```

File: packages/rrweb-snapshot/package.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  },
+  "homepage": "https://github.com/rrweb-io/rrweb-snapshot#readme",
```
Comment (Juice10) [inline packages/rrweb-snapshot/package.json:36]:
```suggestion
  "homepage": "https://github.com/rrweb-io/rrweb#readme",
```

File: packages/rrweb-player/package.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    "type": "git",
+    "url": "git+https://github.com/rrweb-io/rrweb-player.git"
+  },
+  "keywords": [
+    "rrweb"
+  ],
+  "author": "yanzhen@smartx.com",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/rrweb-io/rrweb-player/issues"
```
Comment (Juice10) [inline packages/rrweb-player/package.json:57]:
```suggestion
    "url": "https://github.com/rrweb-io/rrweb/issues"
```

File: packages/rrweb-player/package.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  "unpkg": "dist/index.js",
+  "files": [
+    "lib",
+    "dist",
+    "typings"
+  ],
+  "typings": "typings/index.d.ts",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/rrweb-io/rrweb-player.git"
```
Comment (Juice10) [inline packages/rrweb-player/package.json:49]:
```suggestion
    "url": "git+https://github.com/rrweb-io/rrweb.git"
```

File: packages/rrweb-player/package.json [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  },
+  "keywords": [
+    "rrweb"
+  ],
+  "author": "yanzhen@smartx.com",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/rrweb-io/rrweb-player/issues"
+  },
+  "homepage": "https://github.com/rrweb-io/rrweb-player#readme"
```
Comment (Juice10) [inline packages/rrweb-player/package.json:59]:
```suggestion
  "homepage": "https://github.com/rrweb-io/rrweb#readme"
```

**Review by Juice10 — CHANGES_REQUESTED** [review]
Looking really good @Mark-Fenng! I found some very minor outdated urls, but apart from that this looks good!

**Review by Juice10 — APPROVED** [review]

**Comment by Yuyz0112** [comment]
Nice! Let's go.

I think contributors may see some conflicts in their existing PRs, we can help them get things done again.

And I'm going to check the rollup things and update the release script.

Awesome work.

**Pull request merged**